slack_sdk/web/async_client.py

```diff
@@ -2992,6 +2995,160 @@ async def files_upload(
             kwargs["content"] = content
             return await self.api_call("files.upload", data=kwargs)
 
+    async def files_upload_v2(
+        self,
+        *,
+        # for sending a single file
+        filename: Optional[str] = None,  # you can skip this only when sending along with content parameter
+        file: Optional[Union[str, bytes, IOBase]] = None,
+        content: Optional[Union[str, bytes]] = None,
+        title: Optional[str] = None,
+        alt_txt: Optional[str] = None,
+        snippet_type: Optional[str] = None,
+        filetype: Optional[str] = None,  # no longer supported
+        # To upload multiple files at a time
+        upload_files: Optional[List[Dict[str, Any]]] = None,
+        channel: Optional[str] = None,
+        channels: Optional[Union[str, Sequence[str]]] = None,  # having n channels is no longer supported
+        initial_comment: Optional[str] = None,
+        thread_ts: Optional[str] = None,
+        **kwargs,
+    ) -> AsyncSlackResponse:
+        """This wrapper method provides an easy way to upload files using the following endpoints:
+        * step1: https://api.slack.com/methods/files.getUploadURLExternal
+        * step2: "https://files.slack.com/upload/v1/..." URLs returned from files.getUploadURLExternal API
+        * step3: https://api.slack.com/methods/files.completeUploadExternal
+        """
+        if file is None and content is None and upload_files is None:
+            raise e.SlackRequestError("Any of file, content, and upload_files must be specified.")
+        if file is not None and content is not None:
+            raise e.SlackRequestError("You cannot specify both the file and the content argument.")
+        if channels is not None and (
+            (isinstance(channels, (list, Tuple)) and len(channels) > 1)
+            or (isinstance(channels, str) and len(channels.split(",")) > 1)
+        ):
+            raise e.SlackRequestError("Sharing files with multiple channels is no longer supported")
+        if filetype is not None:
+            warnings.warn("filetype is no longer supported. Please remove it from the arguments.")
```

Comment:
I hesitated to raise an exception for this pattern but if others strongly believe that we should do, I'm fine to change this. But I don't think this can be so beneficial particularly for the devs who migrate existing code.

Reply:
thanks, this is fine